libs/filer/workspace_files_extensions_client.go

```diff
@@ -0,0 +1,328 @@
+			"path":               path.Join(w.root, name),
+			"return_export_info": "true",
+		},
+		stat,
+	)
+	return stat, err
+}
+
+// This function returns the stat for the provided notebook. The stat object itself
+// contains the path with the extension since it is meant used in the context of a fs.FileInfo.
+func (w *workspaceFilesExtensionsClient) removeNotebookExtension(ctx context.Context, name string) (stat *workspaceFileStatus, ok bool) {
+	ext := path.Ext(name)
+
+	nameWithoutExtension := strings.TrimSuffix(name, ext)
+
+	// File name does not have an extension associated that is with Databricks
+	// notebooks, return early.
+	if _, ok := extensionsToLanguages[ext]; !ok {
+		return nil, false
+	}
+
+	// If the file could be a notebook, check if it is and has the correct language.
+	stat, err := w.stat(ctx, nameWithoutExtension)
+	if err != nil {
+		log.Debugf(ctx, "attempting to determine if %s could be a notebook. Failed to fetch the status of object at %s: %s", name, path.Join(w.root, nameWithoutExtension), err)
+		return nil, false
+	}
+
+	// Not a notebook. Return early.
+	if stat.ObjectType != workspace.ObjectTypeNotebook {
+		log.Debugf(ctx, "attempting to determine if %s could be a notebook. Found an object at %s but it is not a notebook. It is a %s.", name, path.Join(w.root, nameWithoutExtension), stat.ObjectType)
+		return nil, false
+	}
+
+	// Not the correct language. Return early.
+	if stat.Language != extensionsToLanguages[ext] {
+		log.Debugf(ctx, "attempting to determine if %s could be a notebook. Found a notebook at %s but it is not of the correct language. Expected %s but found %s.", name, path.Join(w.root, nameWithoutExtension), extensionsToLanguages[ext], stat.Language)
+		return nil, false
+	}
+
+	// When the extension is .py we expect the export format to be source.
+	// If it's not, return early.
+	if ext == ".py" && stat.ReposExportFormat != workspace.ExportFormatSource {
+		log.Debugf(ctx, "attempting to determine if %s could be a notebook. Found a notebook at %s but it is not exported as a source notebook. Its export format is %s.", name, path.Join(w.root, nameWithoutExtension), stat.ReposExportFormat)
+		return nil, false
+	}
+
+	// When the extension is .ipynb we expect the export format to be Jupyter.
+	// If it's not, return early.
+	if ext == ".ipynb" && stat.ReposExportFormat != workspace.ExportFormatJupyter {
+		log.Debugf(ctx, "attempting to determine if %s could be a notebook. Found a notebook at %s but it is not exported as a Jupyter notebook. Its export format is %s.", name, path.Join(w.root, nameWithoutExtension), stat.ReposExportFormat)
+		return nil, false
+	}
+
+	// Modify the stat object path to include the extension. This stat object will be used
+	// to return the fs.FileInfo object in the stat method.
+	stat.Path = stat.Path + ext
+	return stat, true
+}
+
+func (w *workspaceFilesExtensionsClient) addNotebookExtension(ctx context.Context, name string) (*workspaceFileStatus, error) {
```

Comment:
Should be renamed to something like `getNotebookStatByNameWithoutExt`

Reply:
Done.